.github/workflows/tfplugindocs.yaml

```diff
@@ -0,0 +1,48 @@
+---
+name: Terraform plugin docs
+concurrency: tfplugindocs
+on:
+    push:
+        branches:
+            - master
+
+permissions:
+    contents: write
+
+jobs:
+    tfplugindocs:
+        name: Generate Terraform plugin docs
+        runs-on: ubuntu-latest
+        steps:
+            - name: Check out code
+              uses: actions/checkout@v4
+
+            - name: Set up Go
+              uses: actions/setup-go@v4
+              with:
+                  go-version-file: go.mod
+
+            - name: Generate docs
+              run: |
+                  make install-tools tfdocs_generate
+
+            - name: Fix trailing whitespace in generated docs
+              uses: pre-commit/action@v3.0.0
+              # ignoring errors, as pre-commit will fail if files were modified
+              continue-on-error: true
+              with:
+                  extra_args: trailing-whitespace --all-files
+
+            - name: Fix end of file in generated docs
+              uses: pre-commit/action@v3.0.0
+              # ignoring errors, as pre-commit will fail if files were modified
+              continue-on-error: true
+              with:
+                  extra_args: end-of-file-fixer --all-files
+
+            - name: Commit generated docs
+              uses: stefanzweifel/git-auto-commit-action@v4
+              with:
+                  # using ci: prefix to hide this commit from changelog
+                  commit_message: "ci: update Terraform plugin docs"
+                  commit_options: '--no-verify --signoff'
```

Comment:
Why you don't use normal `git add .` and `git commit -m"" --no-verify --signoff` without any dependency? Just asking

Reply:
Because I don't want to deal with:
- Adding git username & email
- Handling non-zero exit code if working tree is clean
- etc.

You can look here for all the different checks & actions: https://github.com/stefanzweifel/git-auto-commit-action/blob/master/entrypoint.sh